### PR TITLE
use jackson 2.9.3

### PR DESCRIPTION
**Pull request by bradbeck**
Remediation for SONATYPE-2017-0355.
http://bamboo.s/browse/OSS-SIESTAF12

### CODE REVIEW

File: siesta-jackson/src/main/java/org/sonatype/sisu/siesta/jackson/ObjectMapperProvider.java [hunk 1]
```diff
@@ -38,10 +37,6 @@ public ObjectMapperProvider() {
     // Configure Jackson annotations only, JAXB annotations can confuse and produce improper content
     mapper.getDeserializationConfig()
         .with(new JacksonAnnotationIntrospector());
-    // do not write null values
-    mapper.getSerializationConfig()
-        .with(new JacksonAnnotationIntrospector())
-        .withSerializationInclusion(Include.NON_NULL);
```
Comment (mcculls) [inline siesta-jackson/src/main/java/org/sonatype/sisu/siesta/jackson/ObjectMapperProvider.java]: AFAICT only the `withSerializationInclusion` method was removed, there's no need to drop the annotation introspector.

It also looks like the new approach to ignore null values when serializing is to use `setSerializationInclusion` which ends up calling `setDefaultPropertyInclusion`, for example:
```
    // Configure Jackson annotations only, JAXB annotations can confuse and produce improper content
    mapper.getDeserializationConfig()
        .with(new JacksonAnnotationIntrospector());
    mapper.getSerializationConfig()
        .with(new JacksonAnnotationIntrospector());

    // do not write null values
    mapper.setSerializationInclusion(Include.NON_NULL);
```

**Comment by mcculls** [comment]
building this locally I find I also need these test plugin/dependency updates:
```
diff --git a/siesta-testsuite/pom.xml b/siesta-testsuite/pom.xml
index dc8e4e5..5f13c32 100644
--- a/siesta-testsuite/pom.xml
+++ b/siesta-testsuite/pom.xml
@@ -68,12 +68,12 @@
       <plugin>
         <groupId>org.jvnet.jaxb2.maven2</groupId>
         <artifactId>maven-jaxb2-plugin</artifactId>
-        <version>0.8.3</version>
+        <version>0.9.1</version>
         <dependencies>
           <dependency>
             <groupId>org.sonatype.sisu.jacksbee</groupId>
             <artifactId>jacksbee-xjc</artifactId>
-            <version>1.1.1</version>
+            <version>1.3</version>
           </dependency>
         </dependencies>
         <configuration>
@@ -83,7 +83,7 @@
             <plugin>
               <groupId>org.sonatype.sisu.jacksbee</groupId>
               <artifactId>jacksbee-xjc</artifactId>
-              <version>1.1.1</version>
+              <version>1.3</version>
             </plugin>
           </plugins>
         </configuration>
```
otherwise I get an error about `Failed to read schema document 'xjc.xsd', because 'file' access is not allowed due to restriction set by the accessExternalSchema property.`

**Comment by bradbeck** [comment]
I was able to build it locally with:
`mvn clean install -Djavax.xml.accessExternalSchema=all`

**Comment by mcculls** [comment]
Generally we prefer to avoid needing custom flags to get a clean build - upgrading those test dependencies will mean you don't need that flag (this is what we did previously for Nexus 2.x when we encountered a similar issue around XJC)

**Comment by wwannemacher** [comment]
@bradbeck - RH suggests calling `objectMapper.enableDefaultTyping()` to mitigate the issue here: https://access.redhat.com/security/cve/cve-2017-15095 

They also suggest using a specific annotation on classes, but I didn't check if we were already doing that... Did you try enabling default typing?

**Comment by wwannemacher** [comment]
@bradbeck - err, darn-it, I misread, it is to *not* call `.enableDefaultTyping`

**Review by wwannemacher — APPROVED** [review]
+1
although, it seems that since we are not using default typing, we may not be vulnerable to this CVE. I think we forged ahead with the dependency update anyway in nxrm3, so I don't have a problem with doing the same here.

File: siesta-jackson/src/main/java/org/sonatype/sisu/siesta/jackson/ObjectMapperProvider.java [hunk 2]
```diff
@@ -40,8 +39,7 @@ public ObjectMapperProvider() {
         .with(new JacksonAnnotationIntrospector());
     // do not write null values
```
Comment (mcculls) [inline siesta-jackson/src/main/java/org/sonatype/sisu/siesta/jackson/ObjectMapperProvider.java]: This comment should be moved/removed as it no longer applies to the line below it.

File: siesta-jackson/src/main/java/org/sonatype/sisu/siesta/jackson/ObjectMapperProvider.java [hunk 3]
```diff
@@ -40,8 +39,7 @@ public ObjectMapperProvider() {
...
     mapper.getSerializationConfig()
-        .with(new JacksonAnnotationIntrospector())
-        .withSerializationInclusion(Include.NON_NULL);
```
Comment (mcculls) [inline siesta-jackson/src/main/java/org/sonatype/sisu/siesta/jackson/ObjectMapperProvider.java]: I did some testing and as it stands this writes out null fields in responses, whereas before those were excluded.

To restore the original behaviour you can add the following line:
```
mapper.setSerializationInclusion(Include.NON_NULL);
```
(note this is now separate from the `serializationConfig` fluent API)

**Review by mcculls — APPROVED** [review]
+1 once excluding nulls from responses has been restored

**Pull request merged**